solr/core/src/test/org/apache/solr/handler/TestContainerPlugin.java

```diff
@@ -366,7 +381,7 @@ public void m2(SolrQueryRequest req, SolrQueryResponse rsp) {
 
   }
 
-  public static class CConfig extends PluginMeta {
+  public static class CConfig implements ReflectMapWriter {
```

Comment:
I don't understand this change. Do all config beans have to implement `ReflectMapWriter`?

If that's the case, and all config beans have to implement ReflectMapWriter then that's what the generic type in ConfigurablePlugin<T> should use. If not then this test is misleading again because it suggests you have to implement additional interfaces to have a usable config bean.

Reply:
All beans must implement `MapWriter`

Reply:
Then we have to explicitly say so in `ConfigurablePlugin<T extends MapWriter>`.